### PR TITLE
Unify section term and interim reports

### PR DESCRIPTION
**Pull request by nbey**
- Unify Section Term Reports with Section Interim Reports

### CODE REVIEW

**Comment by themightychris** [comment]
@nbey this PR is setup to merge a feature into `develop`, it's not a release PR

**Comment by nbey** [comment]
Ok, i'll rename it, and create a new PR accordingly.

File: sencha-workspace/SlateAdmin/app/model/course/SectionTermData.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    ],
+
+
+    // model config
+    idProperty: 'ID',
+    identifier: 'negative',
+
+    fields: [
+        {
+            name: "ID",
```
Comment (themightychris) [inline sencha-workspace/SlateAdmin/app/model/course/SectionTermData.js]: @nbey replace all these double quotes with single quotes, your linter should be warning about this if it is configured correctly

File: sencha-workspace/SlateAdmin/app/controller/progress/terms/Report.js [hunk 2]
```diff
@@ -1,57 +1,61 @@
...
 
     stores: [
         'Terms',
-        'progress.narratives.Sections',
-        'progress.narratives.Students',
-        'progress.narratives.Reports'
+        'progress.terms.Sections',
+        'progress.terms.Students',
+        'progress.terms.Reports',
+        'courses.SectionTermData'
```
Comment (themightychris) [inline sencha-workspace/SlateAdmin/app/controller/progress/terms/Report.js]: @nbey a store should not be used for loading/saving a single record, only managing a population of records. Use the static `.load()` method on the model to make a request to load an individual model without a store. I believe you can pass null for id and then use the options param to pass arbitrary params for the load request.

**Review by themightychris — CHANGES_REQUESTED** [review]
@nbey the interims section also needs section note loading/saving implemented.

It looks like the interims controller has a bunch of incomplete nonfunctional code towards managing section notes that appears to be have copied from the old Narratives controller, which demonstrates the preferred way to load/save individual models: https://github.com/SlateFoundation/slate-admin/blob/3748ef3a2234a1d7ed69092bb7f6fa3ad54512da/sencha-workspace/SlateAdmin/app/controller/progress/Narratives.js#L184

Why did you remove the individual model loading/saving in favor of using a store?

Also, there is some feedback that needs to be addressed in SlateFoundation/slate-core-data#4 -- please push commits there and bump the submodule version in this PR

**Comment by nbey** [comment]
@themightychris Could you check out the latest commits I made?

File: sencha-workspace/SlateAdmin/app/controller/progress/interims/Report.js [hunk 3]
```diff
@@ -220,12 +230,39 @@ Ext.define('SlateAdmin.controller.progress.interims.Report', {
...
         studentsStore.load();
 
         reportsProxy.setExtraParam('term', termHandle);
         reportsProxy.setExtraParam('course_section', sectionCode);
         reportsStore.load();
+
+        sectionNotesForm.enable();
+        sectionNotesForm.setLoading('Loading notes&hellip;');
+
+        SectionNotesModel.getProxy().createOperation('read', {
```
Comment (themightychris) [inline sencha-workspace/SlateAdmin/app/controller/progress/interims/Report.js]: @nbey this name needs updating to `SectionTermDataModel`

File: sencha-workspace/SlateAdmin/app/controller/progress/interims/Report.js [hunk 4]
```diff
@@ -220,12 +230,39 @@ Ext.define('SlateAdmin.controller.progress.interims.Report', {
...
+            },
+            callback: function(sectionNotes, operation, success) {
+                if (!success) {
+                    return;
+                }
+
+                if (!(sectionNotes = sectionNotes[0])) {
+                    sectionNotes = new SectionNotesModel({
+                        TermID: term.getId(),
+                        CourseSectionID: section.getId()
```
Comment (themightychris) [inline sencha-workspace/SlateAdmin/app/controller/progress/interims/Report.js]: @nbey creating new section notes does not work currently...below on line 351 we're getting `sectionDataRecord.get('SectionID')` and aborting if it doesn't resolve to a section, but here we're setting `sectionNotes.get('CourseSectionID')`

File: sencha-workspace/SlateAdmin/app/controller/progress/interims/Report.js [hunk 5]
```diff
@@ -310,21 +347,26 @@ Ext.define('SlateAdmin.controller.progress.interims.Report', {
 
     onSectionNotesSaveBtnClick: function() {
         var sectionNotesForm = this.getSectionNotesForm(),
-            sectionNotes = sectionNotesForm.getRecord();
+            sectionDataRecord = sectionNotesForm.getRecord(),
```
Comment (themightychris) [inline sencha-workspace/SlateAdmin/app/controller/progress/interims/Report.js:362]: @nbey rename to `sectionTermDataRecord` throughout for consistency, these models can get pretty confusing to follow if we're not absolutely consistent with naming them

File: sencha-workspace/SlateAdmin/app/controller/progress/interims/Report.js [hunk 6]
```diff
@@ -310,21 +347,26 @@ Ext.define('SlateAdmin.controller.progress.interims.Report', {
...
+
+        if (!sectionDataRecord.dirty) {
             return;
         }
 
         sectionNotesForm.setLoading('Saving notes&hellip;');
-        sectionNotes.save({
-            callback: function(sectionNotes, operation, success) {
+        sectionDataRecord.save({
+            callback: function(sectionData, operation, success) {
```
Comment (themightychris) [inline sencha-workspace/SlateAdmin/app/controller/progress/interims/Report.js:377]: @nbey call this `savedSectionTermDataRecord`. It should be the same instance as `sectionTermDataRecord` but we need to receive the param and linters don't like the same variable being redeclared. Theoretically the save method COULD return a new model instance, so all the code within this callback should use the potentially-new `saved*` instance reference

**Review by themightychris — CHANGES_REQUESTED** [review]
comments left for interims probably apply equally for terms

File: sencha-workspace/SlateAdmin/app/view/progress/SectionNotesForm.js [hunk 7]
```diff
@@ -1,11 +1,15 @@
 /*jslint browser: true, undef: true, white: false, laxbreak: true *//*global Ext,Slate*/
```
Comment (themightychris) [inline sencha-workspace/SlateAdmin/app/view/progress/SectionNotesForm.js]: @nbey strip this jslint config

**Comment by themightychris** [comment]
@nbey one last thing, let's add responsibilities overviews to each of the progress interims/terms controllers like these have:

- https://github.com/JarvusInnovations/spark-classroom/blob/develop/sencha-workspace/SparkClassroomTeacher/app/controller/Viewport.js
- https://github.com/JarvusInnovations/spark-classroom/blob/develop/sencha-workspace/SparkClassroomTeacher/app/controller/Work.js

**Pull request merged**